oslabs-beta/reactime
# Fixes bugs related to first state and circular fiber trees

## Description:
- Adds fiberRoot in first state; fixed circular tree node problems in tree creation and tree copy.

## Changes I Made:
- Now fiberRoot is available for all websites at start without using npm package.
- Circular nodes are skipped when generating a new snapshot tree and when creating copies of the snapshot tree, avoiding infinite loops.
- Added debug flag in linkFiber.js for omitting console.logs on large websites for performance while developing by overwriting console.log function. Added alwaysLog function that bypasses flag (use judiciously).
- Improved performance by removing 'stateless' components. Still need to find better solution for large websites.

## How to Test:
- Open Reactime in codesmith.io. Now you can see state of the site and perform some time-travel.
